source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -25,1139 +25,3368 @@
+    #error FreeRTOSConfig.h has not been included yet
 #endif
 
-/* Through time some macro names have changed. This always
- * happened for a good reason: clarity or consistency.
- * Here below there are some ifdef's that will issue an error if any of the
- * deprecated macros is still defined. */
-
-/* Ensure defined configuration constants are using the most up to date naming. */
-#ifdef tcpconfigIP_TIME_TO_LIVE
-    #error now called: ipconfigTCP_TIME_TO_LIVE
+#ifndef FREERTOS_IP_CONFIG_H
+    #error FreeRTOSIPConfig.h has not been included yet
 #endif
 
-#ifdef updconfigIP_TIME_TO_LIVE
-    #error now called: ipconfigUDP_TIME_TO_LIVE
-#endif
+#include "FreeRTOSIPDeprecatedDefinitions.h"
 
-#ifdef ipFILLER_SIZE
-    #error now called: ipconfigPACKET_FILLER_SIZE
-#endif
+#include <stdint.h>
 
-#ifndef ipconfigENDPOINT_DNS_ADDRESS_COUNT
-    #define ipconfigENDPOINT_DNS_ADDRESS_COUNT    2U
-#endif
+/*===========================================================================*/
+/*---------------------------------------------------------------------------*/
+/*===========================================================================*/
+/*                                  MACROS                                   */
+/*===========================================================================*/
 
-#ifdef dnsMAX_REQUEST_ATTEMPTS
-    #error now called: ipconfigDNS_REQUEST_ATTEMPTS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigUDP_TASK_PRIORITY
-    #error now called: ipconfigIP_TASK_PRIORITY
-#endif
+/*
+ * Used to define away static keyword for CBMC proofs
+ */
 
-#ifdef ipconfigUDP_TASK_STACK_SIZE_WORDS
-    #error now called: ipconfigIP_TASK_STACK_SIZE_WORDS
+#ifndef _static
+    #define _static    static
 #endif
 
-#ifdef ipconfigDRIVER_INCLUDED_RX_IP_FILTERING
-    #error now called: ipconfigETHERNET_DRIVER_FILTERS_PACKETS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigMAX_SEND_BLOCK_TIME_TICKS
-    #error now called: ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS
-#endif
+/*
+ * Since all code is made compatible with the MISRA rules, the inline functions
+ * disappear. Normally defined in portmacro.h or FreeRTOSConfig.h.
+ */
 
-#ifdef ipconfigUSE_RECEIVE_CONNECT_CALLBACKS
-    #error now called: ipconfigUSE_CALLBACKS
+#ifndef portINLINE
+    #define portINLINE    inline
 #endif
 
-#ifdef ipconfigNUM_NETWORK_BUFFERS
-    #error now called: ipconfigNUM_NETWORK_BUFFER_DESCRIPTORS
-#endif
+/*---------------------------------------------------------------------------*/
 
-#ifdef ipconfigTCP_HANG_PROT
-    #error now called: ipconfigTCP_HANG_PROTECTION
-#endif
+/*
+ * Used to standardize macro checks since ( MACRO == 1 ) and ( MACRO != 0 )
+ * are used inconsistently.
+ */
 
-#ifdef ipconfigTCP_HANG_PROT_TIME
-    #error now called: ipconfigTCP_HANG_PROTECTION_TIME
+#ifndef ipconfigENABLE
+    #define ipconfigENABLE    ( 1 )
```

Comment:
With this are we trying to remove all 0 and 1 check and use ipconfigENABLE & ipconfigDISABLE?

Reply:
I just did that to maintain a standardized 'enable' value in the config files since (x == 1) and (x != 0) are used randomly throughout the source and aren't equivalent statements. But at least with the initial config value checks now you can make sure somebody doesn't put something like 2 when you are expecting it to be either 0 or 1, which would cause inconsistencies in the truth value of the config checks in the source.

Reply:
Yeah it would be good if we can use similar kind of check in the complete TCP code, but that we can take up as a separate effort.